src/git/remote_branch_cache.go

```diff
@@ -1,20 +1,30 @@
 package git
 
+import "errors"
+
 // RemoteBranchCache caches the known remote branches of a Git repository.
 // The zero value is valid.
 type RemoteBranchCache struct {
 	branches    []string
 	initialized bool
 }
 
-// Initialized indicates whether this RemoteBranchTracker is initialized.
+// Get provides the currently known remote branches.
+func (rbt *RemoteBranchCache) Get() ([]string, error) {
+	if !rbt.initialized {
+		return rbt.branches, errors.New("not initialized")
+	}
+	return rbt.branches, nil
+}
+
+// Initialized stores the given remote branches.
 func (rbt *RemoteBranchCache) Initialized() bool {
 	return rbt.initialized
 }
 
-// Get provides the currently known remote branches.
-func (rbt *RemoteBranchCache) Get() []string {
-	return rbt.branches
+// Reset stores the given remote branches.
+func (rbt *RemoteBranchCache) Reset(branches []string) {
```

Comment:
Why are branches passed in here (as they don't appear to be used) and the comment seems wrong.

Reply:
That was an omission, got cleaned up in #1555